aopy/data.py

```diff
@@ -494,4 +572,64 @@ def load_bmi3d_root_metadata(data_dir, filename):
         dict: key-value attributes
     '''
     with h5py.File(os.path.join(data_dir, filename), 'r') as f:
-        return dict(f['/'].attrs.items())
\ No newline at end of file
+        return dict(f['/'].attrs.items())
+
+_cached_signal_path = {}
+
+def lookup_acq2elec(data_dir, signal_path_file, acq, acq_name='acq', elec_name='electrode', zero_index=True):
+    '''
+    Looks up the electrode number for a given acquisition channel using an excel map file (from Dr. Map)
+
+    Inputs:
+        data_dir (str): where the signal path file is located
+        signal_path_file (str): signal path definition file
+        acq (int): which channel to look up
+        acq_name (str, optional): the name of the acquisition column
+        elec_name (str, optional): the name of the electrode column
+        zero_index (bool, optional): use 0-indexing for acq and elec (default True)
+
+    Output:
+        ch2elec [nchannels]: lookup table mapping channels to electrodes or -1
+        pos (nelecs, nelecs): tuple of x and y positions for each electrode
+    '''
+    fullfile = os.path.join(data_dir, signal_path_file)
+    if fullfile in _cached_signal_path:
+        signal_path = _cached_signal_path[fullfile]
+    else:
+        signal_path = pd.read_excel(fullfile)
+        _cached_signal_path[fullfile] = signal_path
```

Comment:
When I tried to run the test code I was missing a package called 'xlrd'. From the error message it looks like it is required to use pd.read_excel so that package may need to be added to setup.py

Reply:
Also, if _cached_signal_path is a list of dataframes it might need a different name.